src/Handler/CurlFactory.php

```diff
@@ -362,9 +362,12 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf)
             if (!is_array($options['proxy'])) {
                 $conf[CURLOPT_PROXY] = $options['proxy'];
             } elseif ($scheme = $easy->request->getUri()->getScheme()) {
-                if (isset($options['proxy'][$scheme])) {
-                    $conf[CURLOPT_PROXY] = $options['proxy'][$scheme];
-                }
+            	$host = $easy->request->getUri()->getHost();
+            	if(!isset($options['proxy']['no']) || !$this->isHostInProxyArea($host, $options['proxy']['no'])) {
```

Comment:
Can yo limit lines to 80 characters and follow PSR-2? http://www.php-fig.org/psr/psr-2/